Client/Assets/SpotlightHelper.js

```diff
@@ -0,0 +1,35 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+  * License, v. 2.0. If a copy of the MPL was not distributed with this
+  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+(function() {
+
+  var selectors = {
+    title: "head title",
+    description: "head meta[name='description'], body p",
```

Comment:
I'm a little worried about this `body p` .. does that mean we grab the full text of the page? I'm worried that can be a lot of data. Isn't the title and the meta description enough to get decent results?